src/core/mount.c

```diff
@@ -923,6 +933,10 @@ static void mount_enter_mounted(Mount *m, MountResult f) {
                 m->result = f;
 
         mount_set_state(m, MOUNT_MOUNTED);
+
+        /* Destroy credentials after successfully mounting. Otherwise, credentials will continue to exist until unit
+         * is stopped. */
+        exec_context_destroy_credentials(&m->exec_context, UNIT(m)->manager->prefix[EXEC_DIRECTORY_RUNTIME], UNIT(m)->id);
```

Comment:
hmm, I am not entirely sure this is desirable. we do allow userspace mount tools to be forked off ExecStart= after all and track them. They might want to continue access the credentials. 

For services we keep the credentials around too all the time, unless we know all processes are gone and there's no ExecStop=. But this doesn't translate well to mount units, because they support umount callouts (i.e. things such as /usr/sbin/umount.nfs are unfortunately a thing that umount calls out to).

Hence I figure we need to keep around the credentials forever.

Reply:
I think this should call the generic unit_destroy_runtime_data() btw, not the more specific exec_context_destroy_credentials()

Reply:
Removed this. Credentials can be deleted with a dependent service unit.